mapbox/io/include/mapbox/io.hpp

```diff
@@ -0,0 +1,73 @@
+#include <string>
+
+#include <ghc/filesystem.hpp>
+#include <nonstd/expected.hpp>
+
+#include <fstream>
+#include <sstream>
+#include <cerrno>
+#include <iostream>
+
+namespace mapbox {
+namespace util {
+
+using ErrorType = std::string;
+
+nonstd::expected<std::string, ErrorType> readFile(const ghc::filesystem::path& filename) {
+    nonstd::expected<std::string, ErrorType> result;
+
+    std::ifstream file(filename.string(), std::ios::binary);
+    if (!file.good()) {
+        return nonstd::make_unexpected(std::string("Failed to open file ") + filename.string());
```

Comment:
Maybe wrap filename in error message in single quotes, i.e.,

    Failed to open file 'foo.txt'

Reply:
Done!